program/tests/stake_instruction.rs

```diff
@@ -0,0 +1,7054 @@
+            &Pubkey::new_unique(),
+        )[0],
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::split_with_seed(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            100,
+            &invalid_stake_state_pubkey(),
+            &Pubkey::new_unique(),
+            "seed",
+        )[1],
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::delegate_stake(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            &invalid_vote_state_pubkey(),
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::withdraw(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            100,
+            None,
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::deactivate_stake(&Pubkey::new_unique(), &Pubkey::new_unique()),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::set_lockup(
+            &Pubkey::new_unique(),
+            &LockupArgs::default(),
+            &Pubkey::new_unique(),
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::deactivate_delinquent_stake(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            &invalid_vote_state_pubkey(),
+        ),
+        Err(ProgramError::IncorrectProgramId),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::deactivate_delinquent_stake(
+            &Pubkey::new_unique(),
+            &invalid_vote_state_pubkey(),
+            &Pubkey::new_unique(),
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::deactivate_delinquent_stake(
+            &Pubkey::new_unique(),
+            &invalid_vote_state_pubkey(),
+            &invalid_vote_state_pubkey(),
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::move_stake(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            100,
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+    process_instruction_as_one_arg(
+        &mollusk,
+        &instruction::move_lamports(
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            &Pubkey::new_unique(),
+            100,
+        ),
+        Err(ProgramError::InvalidAccountData),
+    );
+}
+
```

Comment:
The test `test_spoofed_stake_accounts` doesn't seem to be here -- is that intentional?

Reply:
yes, that test tries to trigger an invalid owner error for every instruction if the first account isnt owned by the program. we dont get the same errors most of the time because this is just an artifact of irregular handling: the first line in every native stake program instruction handler is `let me = get_stake_account()?;` which overrides all other errors like signer checks, checks on other accounts, etc

Reply:
Gotcha, makes sense, thanks!